nixos/modules/installer/cd-dvd/channel.nix

```diff
@@ -31,6 +31,7 @@ let
 in
 
 {
+  nix.registry.nixpkgs.flake = (import (pkgs.path + "/flake.nix")).outputs { self.outPath = pkgs.path; };
```

Comment:
If we had `callLocklessFlake` we would essentially do this:
```nix
{
  nix.registry.nixpkgs.flake = let
    self = { outPath = pkgs.path; } // (import (pkgs.path + "/flake.nix")).outputs { inherit self; };
  in self;
}
```
But I get this error then:
```
$ nix-build nixos/release.nix -A iso_minimal
error: A definition for option `nix.registry.nixpkgs.to.path' is not of type `string or signed integer or boolean or package'. Definition values:
       - In `/home/louis/perso/nixpkgs/master/nixos/modules/services/misc/nix-daemon.nix': /home/louis/perso/nixpkgs/master
```
So we should use `builtins.path` to make it pure:
```nix
{
  nix.registry.nixpkgs.flake = let
    self = { outPath = builtins.path { name = "nixpkgs"; path = pkgs.path; }; } // (import (pkgs.path + "/flake.nix")).outputs { inherit self; };
  in self;
}
```
At which point it appears to eval correctly. Now [since `flake` is a shortcut anyways](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/nixos/modules/services/misc/nix-daemon.nix#L434), we can conclude that it's equivalent to this:
```suggestion
  nix.registry.nixpkgs.to = {
    type = "path";
    path = builtins.path { name = "nixpkgs"; path = pkgs.path; };                                                                                             
  };
```
[I already use a similar trick in my config but I don't know if there's a better way.](https://github.com/lourkeur/config/blob/59fb205daf177056ef98481040540c446817d168/nixos/profiles/misc/link-channels.nix#L17)

Or we could go the imperative route. What do you think?

Reply:
This suggestion is even better https://github.com/NixOS/nixpkgs/pull/189113#discussion_r979430317